test/unit/model/result/operations/replace/Service/LtiReplaceResultParserProxyTest.php

```diff
@@ -0,0 +1,130 @@
+    private $accessTokenRequestValidationResultMock;
+
+    /** @var RegistrationInterface|MockObject */
+    private $registrationMock;
+
+    /** @var ReplaceResultOperationRequest|MockObject */
+    private $replaceResultOperationRequestMock;
+
+    protected function setUp(): void
+    {
+        $this->subject = new LtiReplaceResultParserProxy();
+        $this->accessTokenRequestValidatorMock = $this->createMock(AccessTokenRequestValidator::class);
+        $this->lti1p1ReplaceResultParser = $this->createMock(Lti1p1ReplaceResultParser::class);
+        $this->lti1p3ReplaceResultParser = $this->createMock(Lti1p3ReplaceResultParser::class);
+        $this->replaceResultOperationRequestMock = $this->createMock(ReplaceResultOperationRequest::class);
+
+        $this->accessTokenRequestValidationResultMock = $this->createMock(AccessTokenRequestValidationResult::class);
+        $this->registrationMock = $this->createMock(RegistrationInterface::class);
+
+        $this->requestMock = $this->createMock(ServerRequestInterface::class);
+
+        $this->subject->setServiceLocator(
+            $this->getServiceLocatorMock(
+                [
+                    Lti1p1ReplaceResultParser::class => $this->lti1p1ReplaceResultParser,
+                    Lti1p3ReplaceResultParser::class => $this->lti1p3ReplaceResultParser,
+                    AccessTokenRequestValidator::class => $this->accessTokenRequestValidatorMock,
+                ]
+            )
+        );
+    }
+
+    public function testParseIsLti1p3(): void
+    {
+        $this->accessTokenRequestValidatorMock
+            ->expects($this->once())
+            ->method('validate')
+            ->willReturn($this->accessTokenRequestValidationResultMock);
+
+        $this->accessTokenRequestValidationResultMock
+            ->expects($this->once())
+            ->method('hasError')
+            ->willReturn(false);
+
+        $this->accessTokenRequestValidationResultMock
+            ->expects($this->once())
+            ->method('getRegistration')
+            ->willReturn($this->registrationMock);
+
+        $this->lti1p3ReplaceResultParser
+            ->expects($this->once())
+            ->method('parse')
+            ->willReturn($this->replaceResultOperationRequestMock);
+
+        $this->subject->parse($this->requestMock);
+    }
+
+    public function testParseIsLti1p1(): void
+    {
+        $this->accessTokenRequestValidatorMock
+            ->expects($this->once())
+            ->method('validate')
+            ->willReturn($this->accessTokenRequestValidationResultMock);
+
+        $this->accessTokenRequestValidationResultMock
+            ->expects($this->once())
+            ->method('hasError')
+            ->willReturn(true);
+
+        $this->lti1p1ReplaceResultParser
+            ->expects($this->once())
+            ->method('parse')
+            ->willReturn($this->replaceResultOperationRequestMock);
+
+        $this->subject->parse($this->requestMock);
```

Comment:
We should assert the response content here.

Reply:
This is mocked return. I can test parse on actual `replaceResultParser`

Reply:
Still, you should check this instance was returned, not other one.